internal/server/handler.go

```diff
@@ -59,10 +66,12 @@ func (s *Server) handler() http.HandlerFunc {
 			w.Header().Set(key, strings.Join(value, ", "))
 		}
 
-		_, err = io.Copy(w, resp.Body)
+		n, err := io.Copy(w, resp.Body)
 		if err != nil {
 			s.logger.Warn("failed to copy response body", mlog.Err(err))
 		}
+		elapsed := float64(time.Since(start)) / float64(time.Second)
+		s.metrics.observeRequest(r.URL.Path, r.Method, installationID, resp.StatusCode, n, elapsed)
```

Comment:
It's better to place these in a defer block. There are couple of places in this method which return early in case of an error. We should not miss logging that in prometheus. Status code can be -1 for those.